api/authenticator.py

```diff
@@ -1979,13 +2005,19 @@ def _authentication_flow_document(self, _db):
             self.password_label,
             self.password_label
         )
-        return dict(
+        flow_doc = dict(
             description=unicode(self.DISPLAY_NAME),
             labels=dict(login=unicode(localized_identifier_label),
                         password=unicode(localized_password_label)),
             inputs = dict(login=login_inputs,
                           password=password_inputs)
         )
+        if self.LOGIN_BUTTON_IMAGE:
+            # TODO: I'm not sure if logo is appropriate for this, since it's a button
+            # with the logo on it rather than a plain logo. Perhaps we should use plain
+            # logos instead.
+            flow_doc["links"] = [dict(rel="logo", href=url_for("static_image", filename=self.LOGIN_BUTTON_IMAGE, _external=True))]
```

Comment:
I do think plain logos would be better. It seems like we can turn them into buttons on the client side. Unless there are background color issues.

For the regular auth document `logo` I insert the actual logo as a `data:` URI. This reduces the load on the server since the client (the library registry for a circ manager, SimplyE for the library registry) almost always wants that logo. If you did this here the `static_image` controller would not be necessary.

Reply:
I agree about plain logos, but I'm not sure if I want to change that now. It was originally done as a button rather than a plain logo because Clever provides a button, and students who use Clever will likely be familiar with the button. The button images are the same ones used by the Open eBooks mobile apps, though the apps still have them hard-coded.

For these images, using the URL means they don't need to be downloaded until the user tries to log in, so the initial request for the auth document can be faster. The client probably wants the images most of the time, but doesn't need them immediately.

Reply:
Eventually SimplyE will be retrieving the circ manager's auth document, and in that case it will not need the logo, so maybe we should eventually switch over to having a controller to serve that logo.